pkg/reconciler/tenancy/clusterworkspacetype/clusterworkspacetype_controller.go

```diff
@@ -185,6 +187,10 @@ func (c *controller) processNextWorkItem(ctx context.Context) bool {
 	// other workers.
 	defer c.queue.Done(key)
 
+	logger := logging.WithQueueKey(klog.FromContext(ctx), key)
+	ctx = klog.NewContext(ctx, logger)
+	logger.Info("processing key")
```

Comment:
V(4)?

Reply:
I remember Stefan suggesting something common here before I left for PTO... let me dig it up.

Reply:
- stopping/starting controller - V(0)
- enqueue key - V(2)
- dequeue key - V(4)
- per-key comments
  - processing - V(1)
  - client calls - V(2)
  - something not done - V(4)
- errors - logger.Error()